src/charm.py

```diff
@@ -2,22 +2,208 @@
 
     def __init__(self, *args):
         super().__init__(*args)
+        self._nssf_container_name = self._nssf_service_name = "nssf"
+        self._nssf_container = self.unit.get_container(self._nssf_container_name)
+        self._nrf_requires = NRFRequires(charm=self, relation_name="fiveg_nrf")
+        self._nssf_metrics_endpoint = MetricsEndpointProvider(self)
+        self._service_patcher = KubernetesServicePatch(
+            charm=self,
+            ports=[
+                ServicePort(name="prometheus-exporter", port=PROMETHEUS_PORT),
+                ServicePort(name="sbi", port=SBI_PORT),
+            ],
+        )
+
+        self.framework.observe(self.on.nssf_pebble_ready, self._on_nssf_pebble_ready)
+        self.framework.observe(self._nrf_requires.on.nrf_available, self._on_nssf_pebble_ready)
+
+    def _on_nssf_pebble_ready(
+        self,
+        event: EventBase,
+    ) -> None:
+        """Handle pebble ready event for NSSF container.
+
+        Args:
+            event (EventBase): Juju event
+        """
+        if not self._nssf_container.can_connect():
+            self.unit.status = WaitingStatus("Waiting for container to start")
+            event.defer()
+            return
+        if not self._relation_created("fiveg_nrf"):
+            self.unit.status = BlockedStatus("Waiting for fiveg_nrf relation")
+            return
+        if not self._nrf_data_is_available:
+            self.unit.status = WaitingStatus("Waiting for NRF data to be available")
+            event.defer()
+            return
+        if not self._nssf_container.exists(path=CONFIG_DIR):
+            self.unit.status = WaitingStatus("Waiting for storage to be attached")
+            event.defer()
+            return
+        content = self._render_config_file(
+            sbi_port=SBI_PORT,
+            nrf_url=self._nrf_requires.nrf_url,
+            nssf_url=self._nssf_hostname,
+        )
+        self._push_config_file(
+            content=content,
+        )
+        self._nssf_container.add_layer("nssf", self._nssf_pebble_layer, combine=True)
+        self.unit.status = ActiveStatus()
+
+    def _render_config_file(
+        self,
+        nssf_url: str,
+        sbi_port: int,
+        nrf_url: str,
+    ):
+        """Renders the NSSF config file.
+
+        Args:
+            nssf_url (str): URL of the NSSF.
+            sbi_port (int): NSSF SBi port.
+            nrf_url (str): URL of the NRF.
+        """
+        jinja2_environment = Environment(loader=FileSystemLoader(CONFIG_TEMPLATE_DIR))
+        template = jinja2_environment.get_template(CONFIG_TEMPLATE_NAME)
+        content = template.render(
+            sbi_port=sbi_port,
+            nrf_url=nrf_url,
+            nssf_url=nssf_url,
+        )
+        return content
+
+    def _push_config_file(
+        self,
+        content: str,
+    ) -> None:
+        """Writes the NSSF config file and pushes it to the container.
```

Comment:
From the docstring it might seem this method is doing two steps when is just pushing to the container, isn't it?
```suggestion
        """Pushes the NSSF config file to the container.
```

Reply:
Fixed